gadget/gadget.go

```diff
@@ -326,6 +327,77 @@ func LoadDiskVolumesDeviceTraits(dir string) (map[string]DiskVolumeDeviceTraits,
 	return mapping, nil
 }
 
+// AllDiskVolumeDeviceTraits takes a mapping of volume name to LaidOutVolume and
+// produces a map of volume name to DiskVolumeDeviceTraits. Since doing so uses
+// DiskVolumeDeviceTraitsForDevice, it will also validate that disk devices
+// identified for the laid out volume are compatible and matching before
+// returning.
+func AllDiskVolumeDeviceTraits(allLaidOutVols map[string]*LaidOutVolume, optsPerVolume map[string]*DiskVolumeValidationOpts) (map[string]DiskVolumeDeviceTraits, error) {
```

Comment:
I didn't notice before, the type name doesn't follow our conventions, we never/shouldn't abbreviate Opts in types, we use "Options" in type names

Reply:
This is changed now in the other PR